edgedb-client/src/credentials.rs

```diff
@@ -19,13 +31,62 @@ pub struct Credentials {
     #[serde(default, skip_serializing_if="Option::is_none")]
     pub tls_cert_data: Option<String>,
     #[serde(default, skip_serializing_if="Option::is_none")]
-    pub tls_verify_hostname: Option<bool>,
+    tls_verify_hostname: Option<bool>, // deprecated
+    pub tls_security: TlsSecurity,
+    #[serde(skip)]
+    pub(crate) file_outdated: bool,
+}
+
+
+#[derive(Deserialize)]
+struct CredentialsCompat<'a> {
+    host: Option<Cow<'a, str>>,
+    #[serde(default="default_port")]
+    port: u16,
+    user: Cow<'a, str>,
+    password: Option<Cow<'a, str>>,
+    database: Option<Cow<'a, str>>,
+    tls_cert_data: Option<Cow<'a, str>>,
+    tls_verify_hostname: Option<bool>,
+    tls_security: Option<TlsSecurity>,
 }
 
+
 fn default_port() -> u16 {
     5656
 }
 
+
+impl Credentials {
+    pub fn new(
```

Comment:
Why this huge constructor? Fields of this structure are public (and they are so, basically because they are explicitly public because they are used in JSON). You can't use `Credentials { ... }` syntax, but you can do `Credentials::default()`, and then adjust fields. See such examples in CLI.

In this crate you can probably use `pub(crate)` and use `Credentials {}` syntax, to make sure all fields are correctly filled. This `new` is ugly and will not scale to the new variables added anyway.